backend/dataall/modules/s3_datasets_shares/aws/s3_client.py

```diff
@@ -43,16 +44,28 @@ def get_bucket_access_point_arn(self, access_point_name: str):
 
     def create_bucket_access_point(self, bucket_name: str, access_point_name: str):
         try:
-            access_point = self._client.create_access_point(
+            self._client.create_access_point(
                 AccountId=self._account_id,
                 Name=access_point_name,
                 Bucket=bucket_name,
             )
         except Exception as e:
             log.error(f'S3 bucket access point creation failed for location {bucket_name} : {e}')
-            raise e
-        else:
-            return access_point['AccessPointArn']
+            if 'AccessPointAlreadyOwnedByYou' not in str(e):
+                raise e
+
+        return self.try_get_bucket_access_point_arn(bucket_name, access_point_name)
+
+    def try_get_bucket_access_point_arn(self, bucket_name: str, access_point_name: str):
+        for attempt in range(5):
```

Comment:
you could use `@retry(retry_on_result=lambda arn: arn is None, stop_max_attempt_number=10, wait_fixed=30000)`